sdk/src/main/java/com/hedera/hashgraph/sdk/EthereumFlow.java

```diff
@@ -0,0 +1,133 @@
+package com.hedera.hashgraph.sdk;
+
+import java.util.Arrays;
+
+import java8.util.concurrent.CompletableFuture;
+
+import java.util.concurrent.TimeoutException;
+import javax.annotation.Nullable;
+
+public class EthereumFlow implements WithExecute<TransactionResponse> {
+    @Nullable
+    private EthereumTransactionData ethereumData;
+
+    @Nullable
+    private FileId callDataFileId;
+
+    @Nullable
+    private Hbar maxGasAllowance;
+
+
+    public EthereumFlow() {
+    }
+
+    @Nullable
+    public EthereumTransactionData getEthereumData() {
+        return ethereumData;
+    }
+
+    public EthereumFlow setEthereumData(byte[] ethereumData) {
+        this.ethereumData = EthereumTransactionData.fromBytes(ethereumData);
+        return this;
+    }
+
+    @Nullable
+    public Hbar getMaxGasAllowance() {
+        return maxGasAllowance;
+    }
+
+    public EthereumFlow setMaxGasAllowance(Hbar maxGasAllowance) {
+        this.maxGasAllowance = maxGasAllowance;
+        return this;
+    }
+
+    private static FileId createFile(byte[] callData, Client client) throws PrecheckStatusException, TimeoutException {
+        try {
+            var transaction = new FileCreateTransaction()
+                    .setContents(Arrays.copyOfRange(callData, 0, Math.min(4096, callData.length)))
+                    .execute(client);
+            var fileId = transaction.getReceipt(client).fileId;
+
+            if (callData.length > 4096) {
```

Comment:
Shouldn't this add as many appends as needed?
What if the file since is 9000? You'd need a Create with 2 appends.

Reply:
Same for `createFileAsync`

Reply:
`FileAppendTransaction` is implicitly chunked; meaning it will create N `FileAppendTransactions` internally when executing the request. The only reason we did the initial first chunk manually is because we didn't want to create an empty file and *always* submit a `FileAppendTransaction`.

Reply:
Ah good to know. No-op then